packages/kiota_serialization_json/lib/src/json_parse_node.dart

```diff
@@ -0,0 +1,233 @@
+part of '../kiota_serialization_json.dart';
+
+class JsonParseNode implements ParseNode {
+  JsonParseNode(this._node);
+
+  final dynamic _node;
+
+  @override
+  ParsableHook? onAfterAssignFieldValues;
+
+  @override
+  ParsableHook? onBeforeAssignFieldValues;
+
+  @override
+  bool? getBoolValue() {
+    return _node == null ? null : bool.tryParse(_node.toString());
+  }
+
+  @override
+  Uint8List? getByteArrayValue() {
+    return _node == null ? null : base64Decode(_node.toString());
+  }
+
+  @override
+  ParseNode? getChildNode(String identifier) {
+    if (_node is Map) {
+      final childNode = _node[identifier];
+      if (childNode != null) {
+        return JsonParseNode(childNode)
+          ..onBeforeAssignFieldValues = onBeforeAssignFieldValues
+          ..onAfterAssignFieldValues = onAfterAssignFieldValues;
+      }
+    }
+
+    return null;
+  }
+
+  @override
+  Iterable<T> getCollectionOfEnumValues<T extends Enum>(
+    EnumFactory<T> factory,
+  ) {
+    final result = <T>[];
+    if (_node is List) {
+      for (final value in _node) {
+        final enumValue = factory(value.toString());
+        if (enumValue != null) {
+          result.add(enumValue);
+        }
+      }
+    }
+    return result;
+  }
+
+  @override
+  Iterable<T> getCollectionOfObjectValues<T extends Parsable>(
+    ParsableFactory<T> factory,
+  ) {
+    final result = <T>[];
+    if (_node is List) {
+      for (final value in _node) {
+        final node = JsonParseNode(value)
+          ..onAfterAssignFieldValues = onAfterAssignFieldValues
+          ..onBeforeAssignFieldValues = onBeforeAssignFieldValues;
+        final objectValue = node.getObjectValue(factory);
+        if (objectValue != null) {
+          result.add(objectValue);
+        }
+      }
+    }
+    return result;
+  }
+
+  @override
+  Iterable<T> getCollectionOfPrimitiveValues<T>() {
+    final result = <T>[];
+    if (_node is List) {
+      _node.forEach((value) => result.add(value as T));
```

Comment:
You just use `cast<T>()` here:

```suggestion
      return _node.cast<T>();
```